### DISCUSSION TITLE
Cast SQL Query Date as AutoDate

### DISCUSSION THREAD

**Discussion by mikesimps** [pocketbase/pocketbase#6203] [Q&A] [answered]
I have a scenario where I would like to use a derived updated field in a view collection:

```SQL
SELECT
   max(updated) updated
FROM ...
```

But the schema that is returned says the field is json, not an autodate
<img width="186" alt="image" src="https://github.com/user-attachments/assets/d1c9ed42-6fa8-4931-a06a-c879d0bf33b2" />

Is there a way to cast this field so that the results will be considered an autodate? I tried casting to all sorts of date formats, but nothing changed the schema.

**Comment by ganigeorgiev** [comment, marked as answer]
It is not possible at the moment.

Depending on your use case, with v0.23+ a hacky workaround could be to attach to the  [`OnCollectionCreateExecute`](https://pocketbase.io/docs/go-event-hooks/#oncollectioncreateexecute) and [`OnCollectionUpdateExecute`](https://pocketbase.io/docs/go-event-hooks/#oncollectionupdateexecute) hooks and rewrite the resolved fields after the system resolver, aka. something like:

```go
viewFieldsRemap := &hook.Handler[*core.CollectionEvent]{
    Func: func(e *core.CollectionEvent) error {
        // manually change the view schema field type
        e.Collection.Fields.RemoveByName("updated")
        e.Collection.Fields.Add(&core.AutodateField{
            Name:     "updated",
            OnCreate: true,
            OnUpdate: true,
        })

        return e.Next()
    },
    Priority: 99999, // runs after the system and other handlers
}
app.OnCollectionCreateExecute("yourViewCollection").Bind(viewFieldsRemap)
app.OnCollectionUpdateExecute("yourViewCollection").Bind(viewFieldsRemap)
```

_(the above is not possible with the JSVM because at the moment there is no support for specifying hook priority with the pb_hooks but it may get implemented in the future)_

**Reply by mikesimps** [reply]
Dang, that's what I thought. Thank you for confirming and providing this workaround.
